File: app/src/main/java/com/muaj/oops/module/ActivityModule.java
package com.muaj.oops.module;

import android.app.Activity;
import android.app.Fragment;

import dagger.Module;
import dagger.Provides;

/**
 * @author muaj
 * @date 2018/5/15
 */
@Module
public class ActivityModule {

    private Activity mContext;

    public ActivityModule(Activity activity) {
        mContext = activity;
    }

    public ActivityModule(Fragment fragment) {
        mContext = fragment.getActivity();
    }

    @Provides
    Activity provideActivity() {
        return mContext;
    }
}

File: app/src/main/java/com/muaj/oops/module/ActivityModule.kt
package com.muaj.oops.module

import android.app.Activity
import android.support.v4.app.Fragment

import dagger.Module
import dagger.Provides

/**
 * Created by muaj on 2018/6/6
 *
 */
@Module
class ActivityModule {

    private var mContext: Activity? = null

    constructor(activity: Activity) {
        mContext = activity
    }

    constructor(fragment: Fragment) {
        mContext = fragment.activity
    }

    @Provides
    internal fun provideActivity(): Activity? {
        return mContext
    }
}
